Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/ParabolicTeleportPointer.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Utilities;
+using Microsoft.MixedReality.Toolkit.Internal.Utilities.Lines.DataProviders;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Pointers
+{
+    [RequireComponent(typeof(ParabolaPhysicalLineDataProvider))]
+    public class ParabolicTeleportPointer : TeleportPointer
+    {
+        [SerializeField]
+        private float minParabolaVelocity = 1f;
+
+        [SerializeField]
+        private float maxParabolaVelocity = 5f;
+
+        [SerializeField]
+        private float minDistanceModifier = 1f;
+
+        [SerializeField]
+        private float maxDistanceModifier = 5f;
+
+        [SerializeField]
+        private ParabolaPhysicalLineDataProvider parabolicLineData;
+
+        #region Monobehaviour Implementation
+
+        protected override void OnValidate()
+        {
+            base.OnValidate();
+            EnsureSetup();
+
+            if (parabolicLineData.LineTransform == transform)
+            {
+                Debug.LogWarning("Missing Parabolic line helper.\nThe Parabolic Teleport Pointer requires an empty GameObject child for calculating the parabola arc.");
```

Comment:
Why are we making the dev create the empty child? Seems we should dynamically create / remove automatically

Reply:
We are dynamically creating the child. I just added the warning here just in case something goes wrong.